src/views/AddPropertyManager/index.jsx

```diff
@@ -39,57 +37,55 @@ const AddPropertyManager = () => {
   const [propertyOptions, setPropertyOptions] = useState([]);
   const [propertySearchResults, setPropertySearchResults] = useState([]);
   const [showAddProperty, setShowAddProperty] = useState(false);
-
-  useMountEffect(() => getProperties());
+  const [propertyRefresh, setPropertyRefresh] = useState(true);
 
   useEffect(() => {
-    let choices = propertyOptions.filter(
-      p => p.description.toLowerCase().includes(propertySearchText.toLowerCase()));
-    setPropertySearchResults(choices);
-  }, [propertySearchText, propertyOptions]);
+    setPropertySearchResults(propertyOptions.filter(p =>
+      p.description.toLowerCase().includes(propertySearchText.toLowerCase())
+    )
+  )}, [propertySearchText, propertyOptions]);
+
+  const propertyOptionFormat = (property) => {
+    return {
+      key: property.id,
+      description: `${property.name}, ${property.address}`
+    }
+  }
 
   const getProperties = () => {
-    context.apiCall('get', `/properties`, {}, {})
-      .then(({ data }) => {
-        let properties = data.properties && data.properties.length > 0
-          ? data.properties.map(property => {
-            return {
-              key: property.id,
-              description: `${property.name}, ${property.address}`
-            };
-          })
-          : data.properties;
-        setPropertyOptions(properties);
-        setPropertySearchResults(properties);
-        setShowAddProperty(false);
-      });
+    if (propertyRefresh) {
+      context.apiCall('get', `/properties`, {}, {})
+        .then(({ data }) => {
+          const properties = data.properties && data.properties.length > 0
+            ? data.properties.map(property => {
+              return propertyOptionFormat(property)
+            })
+            : [];
+          setPropertyOptions(properties);
+          setPropertySearchResults(properties);
+          setPropertyRefresh(false)
+        });
+    }
   };
 
   const handleFormSubmit = (data) => {
-    let body = {
+    const payload = {
       ...data,
-      password: 'changeAt1stLogin!',
-      confirmPassword: 'changeAt1stLogin!',
-      type: UserType.PROPERTY_MANAGER,
```

Comment:
Don't we need the "type" parameter since we are moving to that from the Role pattern?

Reply:
Not anymore. The endpoint this uses was changed to `context.apiCall('post', `/user/invite/property_manager`` which only creates a property manager.